sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,231 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PINMUX_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PINMUX_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+// TODO - should be autogenerated?
+#define DIF_PINMUX_MIO_NUM 32
+
+/**
+ * Pinmux peripheral input selects.
+ *
+ * These select which Pad Control MIO output is connected to the requested
+ * peripheral input (MIO output to pinmux input: MIO -> peripheral).
+ */
+typedef enum dif_pinmux_insel {
+  kDifPinmuxInselTieToZero = 0,
+  kDifPinmuxInselTieToOne = 1,
+  kDifPinmuxInselPadFirst = 2,
+  kDifPinmuxInselPadLast = kDifPinmuxInselTieToOne + DIF_PINMUX_MIO_NUM,
+} dif_pinmux_insel_t;
+#define DIF_PINMUX_INSEL_NUM (kDifPinmuxInselPadLast + 1)
+
+/**
+ * Pinmux peripheral output selects.
+ *
+ * These select which MIO input is connected to the requested peripheral output
+ * (pinmux output to MIO input: peripheral -> MIO).
+ * TODO - should be autogenerated?
+ */
+typedef enum dif_pinmux_outsel {
+  kDifPinmuxOutselTieToZero = 0,
+  kDifPinmuxOutselTieToOne = 1,
+  kDifPinmuxOutselHighZ = 2,
+  kDifPinmuxOutselOutGpio0,
+  kDifPinmuxOutselOutGpio1,
+  kDifPinmuxOutselOutGpio2,
+  kDifPinmuxOutselOutGpio3,
+  kDifPinmuxOutselOutGpio4,
+  kDifPinmuxOutselOutGpio5,
+  kDifPinmuxOutselOutGpio6,
+  kDifPinmuxOutselOutGpio7,
+  kDifPinmuxOutselOutGpio8,
+  kDifPinmuxOutselOutGpio9,
+  kDifPinmuxOutselOutGpio10,
+  kDifPinmuxOutselOutGpio11,
+  kDifPinmuxOutselOutGpio12,
+  kDifPinmuxOutselOutGpio13,
+  kDifPinmuxOutselOutGpio14,
+  kDifPinmuxOutselOutGpio15,
+  kDifPinmuxOutselOutGpio16,
+  kDifPinmuxOutselOutGpio17,
+  kDifPinmuxOutselOutGpio18,
+  kDifPinmuxOutselOutGpio19,
+  kDifPinmuxOutselOutGpio20,
+  kDifPinmuxOutselOutGpio21,
+  kDifPinmuxOutselOutGpio22,
+  kDifPinmuxOutselOutGpio23,
+  kDifPinmuxOutselOutGpio24,
+  kDifPinmuxOutselOutGpio25,
+  kDifPinmuxOutselOutGpio26,
+  kDifPinmuxOutselOutGpio27,
+  kDifPinmuxOutselOutGpio28,
+  kDifPinmuxOutselOutGpio29,
+  kDifPinmuxOutselOutGpio30,
+  kDifPinmuxOutselOutGpio31,
+  kDifPinmuxOutselOutLast = kDifPinmuxOutselOutGpio31,
+} dif_pinmux_outsel_t;
+#define DIF_PINMUX_OUTSEL_NUM (kDifPinmuxOutselOutLast + 1)
```

Comment:
This seems like duplication, why have two ways of determining the last value? I thought we'd standardized on one in an earlier DIF review.

Reply:
Done.